game/src/main/kotlin/jake2/game/FuncDoor.kt

```diff
@@ -104,3 +113,94 @@ val funcDoor = registerThink("func_door") { ent, gameExports ->
         ent.think.action = GameFunc.Think_SpawnDoorTrigger
     true
 }
+
+val funcDoorRotating = registerThink("func_door_rotating") { ent, gameExports ->
+    Math3D.VectorClear(ent.s.angles)
+
+    // set the axis of rotation
+    Math3D.VectorClear(ent.movedir)
+    if (ent.spawnflags and GameFunc.DOOR_X_AXIS != 0) ent.movedir[2] =
+        1.0f else if (ent.spawnflags and GameFunc.DOOR_Y_AXIS != 0) ent.movedir[0] = 1.0f else  // Z_AXIS
+        ent.movedir[1] = 1.0f
+
+    // check for reverse rotation
+    if (ent.spawnflags and GameFunc.DOOR_REVERSE != 0) Math3D.VectorNegate(ent.movedir, ent.movedir)
+
+    if (0 == ent.st.distance) {
+        gameExports.gameImports.dprintf("${ent.classname} at ${Lib.vtos(ent.s.origin)} with no distance set")
+        ent.st.distance = 90
+    }
+
+    VectorCopy(ent.s.angles, ent.pos1)
+    VectorMA(
+        ent.s.angles, ent.st.distance.toFloat(), ent.movedir,
+        ent.pos2
+    )
+    ent.moveinfo.distance = ent.st.distance.toFloat()
+
+    ent.movetype = GameDefines.MOVETYPE_PUSH
+    ent.solid = Defines.SOLID_BSP
+    gameExports.gameImports.setmodel(ent, ent.model)
+
+    ent.blocked = GameFunc.door_blocked
+    ent.use = GameFunc.door_use
+
+    if (0f == ent.speed) ent.speed = 100f
+    if (0f == ent.accel) ent.accel = ent.speed
+    if (0f == ent.decel) ent.decel = ent.speed
+
+    if (0f == ent.wait) ent.wait = 3f
+    if (0 == ent.dmg) ent.dmg = 2
+
+    if (ent.sounds != 1) {
+        ent.moveinfo.sound_start = gameExports.gameImports
+            .soundindex("doors/dr1_strt.wav")
+        ent.moveinfo.sound_middle = gameExports.gameImports
+            .soundindex("doors/dr1_mid.wav")
+        ent.moveinfo.sound_end = gameExports.gameImports
+            .soundindex("doors/dr1_end.wav")
+    }
+
+    // if it starts open, switch the positions
+    if (ent.spawnflags and GameFunc.DOOR_START_OPEN != 0) {
+        VectorCopy(ent.pos2, ent.s.angles)
+        VectorCopy(ent.pos1, ent.pos2)
+        VectorCopy(ent.s.angles, ent.pos1)
+        Math3D.VectorNegate(ent.movedir, ent.movedir)
+    }
+
+    if (ent.health != 0) {
+        ent.takedamage = Defines.DAMAGE_YES
+        ent.die = GameFunc.door_killed
+        ent.max_health = ent.health
+    }
+
+    if (ent.targetname != null && ent.message != null) {
+        gameExports.gameImports.soundindex("misc/talk.wav")
+        ent.touch = GameFunc.door_touch
+    }
+
+    ent.moveinfo.state = GameFunc.STATE_BOTTOM
+    ent.moveinfo.speed = ent.speed
+    ent.moveinfo.accel = ent.accel
+    ent.moveinfo.decel = ent.decel
+    ent.moveinfo.wait = ent.wait
+    VectorCopy(ent.s.origin, ent.moveinfo.start_origin)
+    VectorCopy(ent.pos1, ent.moveinfo.start_angles)
+    VectorCopy(ent.s.origin, ent.moveinfo.end_origin)
+    VectorCopy(ent.pos2, ent.moveinfo.end_angles)
+
+    if (ent.spawnflags and 16 != 0) ent.s.effects = ent.s.effects or Defines.EF_ANIM_ALL
```

Comment:
This is confusing without parenthesis. I Initially read it as `flags and (16 !=0 )`

Had to look up that the `and` is actually a bitwise operation